src/hocon.erl

```diff
@@ -50,6 +60,19 @@ proplists(Iter, Path, Acc) ->
             Acc
     end.
 
+convert_value(L, Map) when is_list(L) ->
+    lists:foldl(fun (E, A) -> do_convert_value(E, A) end, Map, L).
```

Comment:
better to create a high-order function which wraps around all the converts and call `nested_update_with` only once.

this is because the `Map` to be updated can be large and each iteration will result in a new map then end up to be garbage collected.

Reply:
nice idea, created `pipeline_fun/1` to handle it.